libs/hdf-converters/src/msft-config-mapper.ts

```diff
@@ -0,0 +1,131 @@
+
+export class MsftConfigMapper extends BaseConverter {
+  profiles: SecureScoreControlProfile[];
+
+  constructor(secureScore_and_profiles_combined: string) {
+    const rawParams = JSON.parse(secureScore_and_profiles_combined);
+
+    super(rawParams.secureScore);
+    this.profiles = rawParams.profiles.value as SecureScoreControlProfile[];
+  }
+
+  mappings: MappedTransform<
+    ExecJSON.Execution & {passthrough: unknown},
+    ILookupPath
+  > = {
+    platform: {
+      name: 'Heimdall Tools',
+      release: HeimdallToolsVersion
+    },
+    version: HeimdallToolsVersion,
+    statistics: {},
+    profiles: [
+      {
+        name: 'Microsoft Secure Score',
+        version: {path: 'v4'},
+        title: {
+          transformer: (d: SecureScore) =>
+            `Azure Secure Score report: TenantID: ${d.azureTenantId}`
+        },
+        summary: 'NAME',
+        supports: [],
+        attributes: [],
+        groups: [],
+        status: 'loaded',
+        controls: [
+          {
+            path: 'controlScores',
+            id: {
+              transformer: (d: ControlScore) =>
+                `${d.controlCategory}:${d.controlName}`
+            },
+            title: {
+              transformer: (d: ControlScore) =>
+                `${d.controlCategory || ''}:${d.controlName || ''}`
+            },
+            desc: {
+              transformer: (d: ControlScore) => d.description || ''
+            },
+            impact: {
+              transformer: (d: ControlScore) => {
+                // return controlCategory from the profile document where its id matches the controlName
+                const knownMaxScores = this.profiles
+                  .filter((p) => p.id === d.controlName)
+                  .map((p) => p.maxScore || 0);
+
+                if (knownMaxScores.length === 0) {
+                  return 0.5;
+                }
+
+                const highMaxScore = Math.max(...knownMaxScores);
+                return highMaxScore / 10.0;
+              }
+            },
+            refs: [],
+            tags: {
+              group: {
+                transformer: (d: ControlScore) => {
+                  // return controlCategory from the profile document where its id matches the controlName
+                  return this.profiles
+                    .filter((p) => p.id === d.controlName)
+                    .map((p) => p.controlCategory);
+                }
+              },
+              threats: {
+                transformer: (d: ControlScore) => {
+                  // return threats from the profile document where its id matches the controlName
+                  return this.profiles
+                    .filter((p) => p.id === d.controlName)
+                    .map((p) => p.threats);
+                }
+              }
+            },
+            source_location: {},
+            results: [
+              {
+                status: {
+                  transformer: (d: ControlScore | any) => {
```

Comment:
Any Typescript gurus, Help needed to allow access to field that is often/always present in the data, but not in the Typescript defs provided by Microsoft.

Reply:
Hey, could you clarify which field you are talking about specifically?

Reply:
`d. scoreInPercentage` exists in the returned JSON but not the typescript type def

Reply:
```
if (d.scoreInPercentage === 100) {
```
It looks like you're already successfully grabbing it here, or am I looking at the wrong place?

Reply:
This was just to address the 'code smell' part of the CI/CD pipeline